kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/xstream/XStreamMarshaller.java

```diff
@@ -103,7 +116,16 @@ public Class realClass(String elementName) {
                 };
             }
         });
+    }
 
+    private MapperWrapper chainMapperWrappers(List<XStreamMarshallerExtension> extensions, MapperWrapper last) {
+        if (extensions.isEmpty()) {
+            return last;
+        } else {
+            XStreamMarshallerExtension head = extensions.get(0);
+            List<XStreamMarshallerExtension> tail = extensions.subList(1, extensions.size());
```

Comment:
looks like we have it now both get and remove in use. 
```
XStreamMarshallerExtension head = extensions.get(0);
 extensions.remove(0);
```

maybe replacing it with 
```
XStreamMarshallerExtension head = extensions.remove(0);
``` 
would be enough...